src/sleep/sleep.c

```diff
@@ -243,32 +232,37 @@ static int execute_s2h(usec_t hibernate_delay_sec) {
         if (r < 0)
                 return r;
 
-        wake_time = original_time + DIV_ROUND_UP(hibernate_delay_sec, USEC_PER_SEC);
-        r = rtc_write_wake_alarm(wake_time);
-        if (r < 0)
-                return r;
+        tfd = timerfd_create(CLOCK_BOOTTIME_ALARM, TFD_NONBLOCK|TFD_CLOEXEC);
+        if (tfd < 0)
+                return log_error_errno(tfd, "Error creating timerfd");
 
-        log_debug("Set RTC wake alarm for %" PRIu64, wake_time);
+        wake_time = DIV_ROUND_UP(hibernate_delay_sec, USEC_PER_SEC);
+
+        log_debug("Set timerfd wake alarm for %"PRIu64 " seconds", wake_time);
```

Comment:
please use format_timespan() to format a nice, human readable string of the timeout